content/blog/2019-11-mio-v0.7-alpha.md

```diff
@@ -0,0 +1,201 @@
++++
+
+
+Mio 0.7 is the work of [various contributors] over the course of roughly half a
+year. Compared to Mio version 0.6 version 0.7 reduces the size of the provided
+API in an attempt to simply the implementation and usage. The API version 0.7
+will be close to the proposed API for a future version 1.0. The scope for the
+crate was reduced to providing a cross-platform event notification mechanism and
+commonly used types such as cross-thread poll waking and non-blocking networking
+I/O primitives.
+
+<!-- TODO: not sure if this is correct timeline. -->
+[various contributors]: https://github.com/tokio-rs/mio/graphs/contributors?from=2019-03-01&to=2019-11-03&type=c
+
+
+## Major changes
+
+Since this is a large release only some highlights are described here, all
+changes can be found in the [change log]. Overall a large number of API changes
+have been made to reduce the complexity of the implementation and remove
+overhead were possible.
+
+<!-- TODO: link to v0.7 in the changelog. -->
+<!-- TODO: also write the changelog. -->
+[change log]: https://github.com/tokio-rs/mio/blob/master/CHANGELOG.md
+
+
+### Wrapping native OS types
+
+In version 0.6 Mio defined it's own `Event` structure to which the native OS
+types where converted. This meant that after polling the native types (i.e.
+`kqueue` or `epoll_event` structures) would be converted into Mio's `Event`
+type. In version 0.7 the `Event` was changed to simply be a type wrapper around
+`kqueue` or `epoll_event` and provides convenience methods to get the event
+readiness indicators such as readable or writable. Similar changes have been
+made throughout the crate, for example `Poll` is now just a file descriptor on
+Unix.
+
+Handling of events in the `Events` has also changed. In version 0.6.10 index
+access was already deprecated and replaced with an iteration API. The index
+access was completely removed in version 0.7 and the iteration API was changed
+to return a reference to `Event` rather then making a copy, but since all
+methods on `Event` only need a reference this shouldn't be a problem.
+
+
+### Removal of the user space queue & deprecated types
+
+Mio version 0.6 had a user space queue, available through the `SetReadiness` and
+`Registration` types, but this has been removed in version 0.7 as it was deemed
+to be outside of the scope for Mio. Users who need a user space queue can use
+one of the queue types found in the [crossbeam] crate.
+
+One use case the user space queue support was waking a thread polling (i.e.
+calling `Poll::poll`) from another thread. To support this use case a new
+`Waker` type was introduced to do exactly this job, calling `Waker::wake` will
+wake the `Poll` it is associated with.
+
+[crossbeam]: https://crates.io/crates/crossbeam
+
+
+### Registering & I/O resource usages changes
+
+By far the biggest user facing change of version 0.7 is the way Mio registers
+I/O sources. In version 0.6 resource would be registered using a `Token`,
```

Comment:
More grammar trouble.  I suggest:

In version 0.6 resources were registered using `Token`, `Ready`, and `PollOpt` arguments.